hedera-node/src/main/java/com/hedera/services/sigs/order/SigReqsManager.java

```diff
@@ -135,50 +128,7 @@ private void expandFromWorkingState(final PlatformTxnAccessor accessor) {
 	 * @return whether the expansion attempt succeeded
 	 */
 	private boolean tryExpandFromSignedState(final PlatformTxnAccessor accessor) {
-		try (final AutoCloseableWrapper<ServicesState> wrapper = platform.getLastCompleteSwirldState()) {
-			final var signedState = wrapper.get();
-			if (signedState == null) {
-				return false;
-			}
-			// Since we can't get the enclosing platform SignedState, we don't know exactly
-			// when this state was signed. So we just use, as a guaranteed lower bound, the
-			// consensus time of its last-handled transaction.
-			final var earliestSigningTime = signedState.getTimeOfLastHandledTxn();
-			if (earliestSigningTime == null) {
-				// No transactions have been handled; abort now to avoid downstream NPE.
-				return false;
-			}
-			if (signedState.getStateVersion() != StateVersions.CURRENT_VERSION) {
-				// We just upgraded and don't yet have a signed state from the current version.
-				return false;
-			}
-			if (!signedState.isInitialized()) {
-				// The aliases FCHashMap has not been rebuilt in an uninitialized state
-				return false;
-			}
-
-			expandFromSignedState(signedState, earliestSigningTime, accessor);
-			return true;
-		}
-	}
-
-	/**
-	 * Tries to expand the platform signatures linked to a transaction from a provided signed state that
-	 * cannot have been signed before the given consensus time. Returns whether the expansion attempt was
-	 * successful. (If this fails, we will next try to expand signatures from the working state.)
-	 *
-	 * @param signedState
-	 * 		the signed state to use for the expansion attempt
-	 * @param earliestSigningTime
-	 * 		the earliest consensus time at which the state could have been signed
-	 * @param accessor
-	 * 		the transaction to expand signatures for
-	 */
-	private void expandFromSignedState(
-			final ServicesState signedState,
-			final Instant earliestSigningTime,
-			final PlatformTxnAccessor accessor
-	) {
+		final SignedStateViewFactory factory = new SignedStateViewFactory(platform);
```

Comment:
Since it is thread-safe, we can make the `SignedStateViewFactory` a Dagger2 `@Singleton` and inject it into the `SigReqsManager`.

Then we can re-use the same instance in our precheck logic when creating the signed-state `StateView`s to be used within a `SignedTxnAccessor`.

Reply:
Do you also think `SignedStateViewFactory` needs a method that can construct `StateView` and return ?